op-service/txmgr/txmgr.go

```diff
@@ -295,18 +293,19 @@ func (m *SimpleTxManager) SendAsync(ctx context.Context, candidate TxCandidate,
 	if err != nil {
 		m.resetNonce()
 		cancel()
-		m.metr.RecordPendingTx(m.pending.Add(-1))
 		ch <- SendResponse{
 			Receipt: nil,
 			Err:     err,
 		}
 		return
 	}
 
+	m.metr.RecordPendingTx(m.pending.Add(1))
+
 	go func() {
-		defer m.metr.RecordPendingTx(m.pending.Add(-1))
 		defer cancel()
 		receipt, err := m.sendTx(ctx, tx)
+		m.metr.RecordPendingTx(m.pending.Add(-1))
```

Comment:
This was indeed the bug, `defer` evaluates all function calls but the outermost of course 😅 
I still think it's worth using `defer` so in case there's a panic in this goroutine, the counter still works.
```suggestion
		defer func() {m.metr.RecordPendingTx(m.pending.Add(-1))}()
```
(and move back to old location)

Reply:
[8c42d46](https://github.com/ethereum-optimism/optimism/pull/14168/commits/8c42d465217e374616c1cea3528193d09112ef32)